src/core/store/modules/app/app.actions.ts

```diff
@@ -34,11 +36,16 @@ const clearUserAppData = createAsyncThunk<void, void, ThunkAPI>('app/clearUserAp
 /*                                 Fetch Data                                 */
 /* -------------------------------------------------------------------------- */
 
-const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState }) => {
+const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState, extra }) => {
+  const { CONTRACT_ADDRESSES } = extra.config;
   const { wallet, network } = getState();
-  if (inIframe()) {
+  if (inLedgerIframe()) {
+    if (network.current !== 'mainnet') await dispatch(NetworkActions.changeNetwork({ network: 'mainnet' }));
```

Comment:
we had problems in the past forcing to change networks, since some wallets doesn't like this behaviour, to have it in mind

Reply:
Yeah, this part changes the selected network for the app, and it only dispatch an event for the wallet network change if its Metamask now. This part its just meant if for some reason ledger live loads in another network (which should not happen), since they just support mainnet.